spec/unit/transaction/additional_resource_generator_spec.rb

```diff
@@ -479,21 +479,6 @@ class container {
                          "Notify[goodbye]"))
     end
 
-    it "sets resources_failed_to_generate to true if resource#generate raises an exception" do
-      catalog = compile_to_ral(<<-MANIFEST)
-        user { 'foo':
-          ensure => present,
-        }
-      MANIFEST
-
-      allow(catalog.resource("User[foo]")).to receive(:generate).and_raise(RuntimeError)
```

Comment:
Not a blocking comment, but I'm thinking this is a useful test, though the way 'generate` was stubbed out previously wasn't helpful. Is there are realistic scenario where `eval_generate` can raise, such as failing to read if the `home` directory is not readable? If so, maybe have `Dir.exists?` raise that error and assert `resources_failed_to_generate` is false?

Reply:
Tried a few ways but couldn't find a valid case where `find_unmanaged_keys`, `munged_unmanaged_keys` or `unknown_keys_in_file` could raise anything. `Dir.exists?` doesn't seem to mind if the folder has any read rights at all so it tells you if the given path exists or not regardless of ownership or rights. `File.readable?` also doesn't raises and since this is used as a pre-filter before trying to open any ssh authorized key files, we can't fail there either.